### ISSUE TITLE
Crm/Contacts/Api/BasicApi.php:getById Doesn't work as described when passed an array

### ISSUE THREAD

**Issue by bensquire** [HubSpot/hubspot-api-php#10]
This method only returns the provided fields when passed a CSV list of items (no space between commas).

Also the docblock describes how the parameters takes string[], but it also just takes string and as such needs updating.

`* @param  string|string[] $properties A comma separated list of the properties to be returned in the response. If any of the specified properties are not present on the requested object(s), they will be ignored. (optional)`

**Comment by jaztec** [comment]
Well, this is kind of an always TRUE when passing an array:

https://github.com/HubSpot/hubspot-api-php/blob/master/codegen/Crm/Contacts/Api/BasicApi.php#L902

I recon the `'form' === 'form'` test should actually test a variable?

**Comment by ksvirkou-hubspot** [comment]
Hi @benbriggs 
Now it should work correctly 
Could you check it on [new version](https://github.com/HubSpot/hubspot-api-php/releases/tag/3.0.2) ?

**Comment by bensquire** [comment]
Apologies in advance (I've long moved jobs since this was raised)

However, if I recall the issue and have found the right file in your new version (https://github.com/HubSpot/hubspot-api-php/blob/7456212a080ec163b11c7849a392fadca6f1c3f4/codegen/Crm/Contacts/Api/BasicApi.php) I think this is still wrong.

`@param  string[] $properties`

implies an array of strings, but the docblock description for that property suggests a CSV (i.e. a string)

It would need changing to:

`@param  string $properties`

which would help people who are using intelligent editors and static analysers to better use the library. It looks like similar issues are also present on other methods.

**Comment by bensquire** [comment]
(Adding method property typing and something like PHPStan/Psalm to check the code, would really improve its usability)

**Comment by jaztec** [comment]
Still `'form' === 'form'` [right here](https://github.com/HubSpot/hubspot-api-php/blob/7456212a080ec163b11c7849a392fadca6f1c3f4/codegen/Crm/Contacts/Api/BasicApi.php#L897).

So something is really borked at the code generating level.

And technically the docblock should state `@param string|string[] $properties`, it works on these conditions and often the first works much better because, well, `'form' === 'form'`.

**Comment by bensquire** [comment]
@jaztec If I understand what you're saying it should infact take an array _or_ a string.

What I don't see (looking at the code) is why `'form' === 'form'` is stopping arrays from being passed in. Since that part of the code just equates to true.

But yes, borked it is

**Comment by jaztec** [comment]
Aight, you're right. `'form' === 'form'` is just logical `true` so that part is sloppiness from the generated code part. And it doesn't stop arrays from being passed in. It does however obscure the array usage from the calling functions.

What happens here is that any string is passed to the `properties` array key and any array needs to define the `properties` key itself and than provide the same `string` from earlier to get the same result as before. The outcome is the same.
The problem really arises [here where calling functions](https://github.com/HubSpot/hubspot-api-php/blob/7456212a080ec163b11c7849a392fadca6f1c3f4/codegen/Crm/Contacts/Api/BasicApi.php#L831) call the property raw.
The calling functions` docblocks indicate that they themselves can be called with an array of strings which, to any consumer, seem like the properties as defined in the official API documentation (body, form, json, query, whatever). Yet it is the exact same set of properties/form-arguments/... the base function will parse any argument to. They can just do the same as the base function and that is passing a string, or an array of strings. But when using an array you are bound to specify the required field (`properties`) and still need to pass the same string.

So, long story short, I think the problem really lies in bad code generation. And technically the base function needs `@param string|string[] $properties` to describe what it' doing :)

**Comment by bensquire** [comment]
@jaztec Thanks for helping me to clarify whats going on in the codebase :)
